torch/onnx/_internal/fx/passes/decomp.py

```diff
@@ -32,6 +37,30 @@ def _rename_placeholder_targets(
     module.recompile()
 
 
+def wrapped_fn(fn, mode, allow_non_fake_inputs, args):
```

Comment:
@bdhirsh @ezyang this is "wrapping" of a callable so that users can trace with an existing fake mode. IMO, it is a lot to expect a user to write, especially considering the number of times we've seen a user attempt to do something like this themselves.... however i'm really just trying to unblock the fake tensor error so I will do whatever to land this

Reply:
I agree we need to work this out a better way to do this. Maybe we can discuss this tomorrow at composability social. My current thinking is that if we can enforce there is one fake mode, it makes sense to have an easy context manager `with_fake` and `no_fake`.